### PR TITLE
ChannelMessageSendComplex

### PR DESCRIPTION
**Pull request by LEGOlord208**
I was getting tired of all the ChannelMessageSendThisWithThisAndThisAndThis.  
So, well, this is my solution.

I kept the others for compatibility. And `ChannelMessageSend` should always be kept for ease of use.

Anyways, I'm unsure whether I should also include `ChannelFileSend` in this as well. It'd certainly be useful (sending embeds and files together, maybe).  
But that's for later.

### CODE REVIEW

File: restapi.go [hunk 1]
```diff
@@ -1332,23 +1324,7 @@ func (s *Session) ChannelMessageSendEmbed(channelID string, embed *MessageEmbed)
 // content   : The message to send.
 // embed     : The embed data to send.
 func (s *Session) ChannelMessageSendEmbedWithMessage(channelID string, content string, embed *MessageEmbed) (st *Message, err error) {
```
Comment (iopred) [inline restapi.go]: Good call, let's kill this method, it was added in develop, I don't care too much to keep it around.

**Comment by iopred** [comment]
We should probably add EditComplex also.

**Comment by LEGOlord208** [comment]
I was going to do that, but Edit doesn't have as many functions and combinations. At all.

File: restapi.go [hunk 2]
```diff
@@ -1303,63 +1305,46 @@ func (s *Session) channelMessageSend(channelID, content string, tts bool) (st *M
...
-
-	// Send the message to the given channel
-	response, err := s.RequestWithBucketID("POST", EndpointChannelMessages(channelID), data, EndpointChannelMessages(channelID))
-	if err != nil {
-		return
-	}
-
-	err = unmarshal(response, &st)
-	return
+func (s *Session) ChannelMessageSendEmbedWithMessage(channelID string, content string, embed *MessageEmbed) (*Message, error) {
```
Comment (iopred) [inline restapi.go]: Let's just kill this guy, WDYT?

Comment (LEGOlord208) [inline restapi.go]: Personally, yes. But it deserves mercy, too. And because I'm the kind of person that likes compatibility.

Comment (iopred) [inline restapi.go]: I love me some compatibility, but this method never made it to master.

Comment (LEGOlord208) [inline restapi.go]: Ok you're right. I will remove them

File: message.go [hunk 3]
```diff
@@ -31,6 +31,20 @@ type Message struct {
 	Reactions       []*MessageReactions  `json:"reactions"`
 }
 
+// MessageSend stores all parameters you can send with ChannelMessageSendComplex.
+type MessageSend struct {
+	Content string        `json:"content"`
+	Tts     bool          `json:"tts"`
+	Embed   *MessageEmbed `json:"embed"`
+	Nounce  string        `json:"nounce"`
```
Comment (iopred) [inline message.go]: This is `nonce`

Comment (LEGOlord208) [inline message.go]: Oohh, right. Thanks. Will fix!

**Pull request merged**